### ISSUE TITLE
[Bug]: mosquitto_passwd Fails Because It Cannot Create A Temporary File

### ISSUE THREAD

**Issue by yasurok** [termux/termux-packages#7357]
### Problem description

The `mosquitto_passwd` command fails when invoked with the `-U` option and when with the `-b` option. In either case, the error message is "Error: Unable to open temporary file. Permission denied." 

Looking at [the source code of mosquitto_passwd](https://github.com/zlargon/mosquitto/blob/master/src/mosquitto_passwd.c), I suspect the failure of the `tmpfile()` function is the culprit. 

The environment variable `TMPDIR` is properly set to `/data/data/com.termux/files/usr/tmp` (I have never changed the value myself). I referenced [this comment from a different thread](https://github.com/termux/termux-app/issues/1550#issuecomment-626200033), and did the following, having absolutely no idea how it can make any difference, but it does not seem to have done the trick, unfortunately.

> SELinux maybe. Try to remove & create tmpdir (as **non-root**):
> 
> ```
> rm -rf $PREFIX/tmp
> mkdir $PREFIX/tmp
> ```

Interestingly,  `mosquitto_passwd` succeeds with the `-c` option. This gave me an opportunity for a makeshift solution where I create a password file for each user and then concatenate all the password files to create one password file, which is then used as the value for `password_file` in the configuration file. It works.

### What steps will reproduce the bug?

Invoke mosquitto_passwd with those options.

### What is the expected behavior?

Just as you can expect from the command option.

### System information

termux-info:

```
Application version:
0.117
Packages CPU architecture:
arm
Subscribed repositories:
# sources.list
deb https://packages.termux.org/apt/termux-main/ stable main
# science-repo (sources.list.d/science.list)
deb https://packages.termux.org/apt/termux-science science stable
# game-repo (sources.list.d/game.list)
deb https://packages.termux.org/apt/termux-games games stable
Updatable packages:
All packages up to date
Android version:
10
Kernel build information:
Linux localhost 4.9.180 #3 SMP PREEMPT Tue Feb 16 14:14:47 UTC 2021 armv8l Android
Device manufacturer:
Amino
Device model:
Amigo7xJP
```

**Comment by MrAdityaAlok** [comment]
@yasurok  could you test whether issue is fixed or not. I'm not sure wheter only this was the issue or not.....

**Comment by yasurok** [comment]
Hi, @MrAdityaAlok. Thank you very much for your prompt attention. I do not know the norm of how fast things get taken care of in the FOSS world, but this was a very nice surprise.

I did install newer packages of mosquitto and libmosquitto. I tried `mosquitto_passwd` with the `-b` option and then with the `-U` option, and it seems to have worked in both cases. I said "seems" because I confirmed only up to the point where a legit-looking password file is created without an error. I did not test if that is accepted by `mosquitto`. 

I appreciate your hard work! Thank you!
